lib/Analysis/InstructionSimplify.cpp

```diff
@@ -3914,8 +3914,15 @@ static Value *SimplifyCall(Value *V, IterTy ArgBegin, IterTy ArgEnd,
   ConstantArgs.reserve(ArgEnd - ArgBegin);
   for (IterTy I = ArgBegin, E = ArgEnd; I != E; ++I) {
     Constant *C = dyn_cast<Constant>(*I);
-    if (!C)
+    if (!C) {
```

Comment:
I think it would be better to call `SimplifyDxilCall` above. From there we can call into constant folding if all operands are constants.